etcdserver/server.go

```diff
@@ -527,6 +539,62 @@ func NewServer(cfg ServerConfig) (srv *EtcdServer, err error) {
 	}
 	srv.r.transport = tr
 
+	// fresh start
+	if !haveWAL {
```

Comment:
why do we need to care about restart vs fresh start?

see https://github.com/coreos/etcd/pull/9364#issuecomment-370096532.

Reply:
Just easier, so that fresh start does not need to synchronize with peer connection reports. But as you suggested, let me simplify the logic (https://github.com/coreos/etcd/pull/9364#issuecomment-370096532).